src/pages/api/iexec/fund-release.tsx

```diff
@@ -0,0 +1,121 @@
+import mongoose from 'mongoose';
+import { getNewPayment } from '../../../queries/payments';
+import { EmailType, IPayment, PaymentTypeEnum, Web3mailPreferences } from '../../../types';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { sendMailToAddresses } from '../../../scripts/iexec/sendMailToAddresses';
+import { getUserWeb3mailPreferences } from '../../../queries/users';
+import { calculateCronData } from '../../../modules/Web3mail/utils/cron-utils';
+import {
+  checkPaymentExistenceInDb,
+  persistCronProbe,
+  persistEmail,
+} from '../../../modules/Web3mail/utils/database-utils';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const chainId = process.env.NEXT_PUBLIC_NETWORK_ID;
+  const mongoUri = process.env.NEXT_MONGO_URI;
+  const cronSecurityKey = req.query.key;
+  const RETRY_FACTOR = 5;
+  let successCount = 0,
+    errorCount = 0;
+
+  if (cronSecurityKey !== process.env.NEXT_PRIVATE_CRON_KEY) {
+    return res.status(401).json({ message: 'Unauthorized' });
+  }
+
+  if (!chainId) {
+    return res.status(500).json('Chain Id is not set');
+  }
+
+  if (!mongoUri) {
+    return res.status(500).json('MongoDb URI is not set');
+  }
+
+  await mongoose.connect(mongoUri as string);
+
+  const platformId = process.env.NEXT_PUBLIC_PLATFORM_ID;
+
+  if (!platformId) {
+    throw new Error('Platform Id is not set');
+  }
+
+  // Check whether the user provided a timestamp or if it will come from the cron config
+  const { sinceTimestamp, cronDuration } = calculateCronData(
+    req,
+    RETRY_FACTOR,
+    EmailType.FundRelease,
+  );
```

Comment:
the ~30 lines above will be common between all API, need to factorize

Reply:
There are 3 different cases actually, they are not common to all APIs